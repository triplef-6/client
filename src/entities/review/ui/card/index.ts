export {Index as Card} from "./Index.tsx"