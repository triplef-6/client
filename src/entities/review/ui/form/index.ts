export {Index as ReviewForm} from "./Index.tsx"