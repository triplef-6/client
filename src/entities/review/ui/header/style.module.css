.container {
    @apply flex flex-col md:flex-row gap-4 md:items-center
}

.heading {
    @apply text-xl text-grayscale-500 font-medium
}