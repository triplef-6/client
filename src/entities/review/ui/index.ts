export {Index as Reviews} from "./Index.tsx"
export * from "./form"