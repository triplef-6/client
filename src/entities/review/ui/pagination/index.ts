export {Index as Pagination} from "./Index.tsx"