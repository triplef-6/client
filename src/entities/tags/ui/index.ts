export {Index as Tag} from "./Index.tsx"