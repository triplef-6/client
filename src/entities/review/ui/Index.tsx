import {FC, useState} from "react";
import {Card} from "./card";
import {Header} from "./header";
import {Pagination} from "./pagination";
import {useReviewsByTourId} from "@/entities";
import {AppSkeleton} from "@/shared/ui";

type ReviewListProps = {
    tourId: number
    rating: number
    ratingCount: number
}

export const Index: FC<ReviewListProps> = ({rating, ratingCount, tourId}) => {

    const {data: reviews, isLoading, isError} = useReviewsByTourId(tourId)
    const [visibleReviews, setVisibleReviews] = useState<number>(2)

    if (isError) return null
    if (isLoading) return <AppSkeleton/>

    return (
        <div className={"flex flex-col gap-8 my-4"}>
            <Header
                rating={rating}
                ratingCount={ratingCount}
            />
            <div className={"flex flex-col gap-4"}>
                {reviews.slice(0, visibleReviews).map(review =>
                    <Card
                        key={review.id}
                        name={review.name}
                        rating={review.rating}
                        positiveText={review.positiveText}
                        negativeText={review.negativeText}
                        withChildren={review.withChildren}
                        personCount={review.personCount}
                    />
                )}
                <Pagination
                    visibleReviews={visibleReviews}
                    setVisibleReviews={setVisibleReviews}
                    reviews={reviews}
                />
            </div>
        </div>
    );
};