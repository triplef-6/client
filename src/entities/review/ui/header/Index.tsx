import {FC} from "react";
import style from "./style.module.css"
import {Rating} from "@/shared/ui"
import {formatReviews} from "@/shared/utills";

type ReviewHeaderProps = {
    rating: number
    ratingCount: number
}

export const Index: FC<ReviewHeaderProps> = ({rating, ratingCount}) => {
    return (
        <div className={style.container}>
            <div className={style.heading}>{formatReviews(ratingCount)}</div>
            <Rating rating={rating} ratingCount={ratingCount}/>
        </div>
    );
};