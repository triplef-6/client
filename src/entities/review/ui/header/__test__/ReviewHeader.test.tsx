import { describe, it, expect } from "vitest"
import { render, screen } from "@testing-library/react"
import {Header} from "@/entities/review/ui/header"

describe("Review header", () => {

    const mockProps = {
        rating: 4.5,
        ratingCount: 44,
    }

    it("Проверка рендеринга", () => {
        render(<Header {...mockProps} />)
    })

    it("Snapshot тест", () => {
        const { container } = render(<Header {...mockProps} />)
        expect(container).toMatchSnapshot()
    })

    it("Проверка отображения количества отзывов", () => {
        render(<Header {...mockProps} />)
        expect(screen.getByText("44 отзыва путешественников")).toBeInTheDocument()
    })

})