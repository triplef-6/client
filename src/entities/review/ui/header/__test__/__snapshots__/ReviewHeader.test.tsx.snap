// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Review header > Snapshot тест 1`] = `
<div>
  <div
    class="_container_4dcacb"
  >
    <div
      class="_heading_4dcacb"
    >
      44
       отзыва путешественников
    </div>
    <div
      class="flex flex-row gap-2"
    >
      <div
        class="flex items-center justify-center w-8 h-6 rounded-xl text-grayscale-0 text-sm bg-secondary-green"
      >
        4.5
      </div>
      <p
        class="text-grayscale-400 text-base"
      >
        На основе 
        44
         оценок
      </p>
    </div>
  </div>
</div>
`;

exports[`ReviewHeader Component > Snapshot тест 1`] = `
<div>
  <div
    class="_container_4dcacb"
  >
    <div
      class="_heading_4dcacb"
    >
      44
       отзыва путешественников
    </div>
    <div
      class="flex flex-row gap-2"
    >
      <div
        class="flex items-center justify-center w-8 h-6 rounded-xl text-grayscale-0 text-sm bg-secondary-green"
      >
        4.5
      </div>
      <p
        class="text-grayscale-400 text-base"
      >
        На основе 
        44
         оценок
      </p>
    </div>
  </div>
</div>
`;
